### PR TITLE
OCPBUGS-32812: When newly built images rolled out, the update progress is not displaying correctly (went 0 --> 3)

### PR DESCRIPTION
**Pull request by djoshy**
**- What I did**

- Tweaked a few things with how OCL populates MCP status numbers. [More details here](https://issues.redhat.com/browse/OCPBUGS-32812?focusedId=25185877&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-25185877).
- Removed MachineConfigNode based status population, this will have to be reworked as a [separate card](https://issues.redhat.com/browse/MCO-1228). 
- Beefed up a few of the node controller log messages, so it is easier to understand what node/pool the controller is working on.
- Updated some of the OCL related unit tests to work with the fixed mechanism, but this is really more of a stopgap. I think a bigger rework needs to be done in the test suite which I believe @cheesesashimi has alluded to. Some of the tests did not seem accurate/correctly setup, but that might be my lack of expertise with OCL.

**- How to verify it**
Follow the [reproduction steps in the bug](https://issues.redhat.com/browse/OCPBUGS-32812), there should not be any odd jumps/inaccuracies in the MCP status numbers for both the layered and non layered MC/image rollouts.

### CODE REVIEW

**Comment by sergiordlr** [comment]
Verified using IPI on AWS

We have checked that the updated nodes are correctly reported in the following scenarios:

- Apply  techpreview
- Apply a new MC in no-techpreview cluster
- Apply OCL images in tech-preview


Nevertheless, we have observed that when applying the OCL image the maxUnavailable value in the MCP is not honored, and we have seen that in pools without any maxUnavailable value (default 1), 2 nodes start applying the OCL image at the same time.

We have tried to reproduce it in other clutsers without this fix. We have only been able to reproduce it once after trying many times. Nonetheless, in the clusters containing this fix this issue is observed basically 100% of the time.

Since we were able to reproduce it once without this fix, we cannot assure that the problem is in this fix, but it seems to be related. Please, could you pay special attention to the behaviour that whe have described here while reviewing this  code? 

If after the review it is decided that this PR has no impact in the behaviour related to maxUnavailable, we can approve this PR and create a different ticket to track the new issue 

Thanks.

**Comment by djoshy** [comment]
> Nevertheless, we have observed that when applying the OCL image the maxUnavailable value in the MCP is not honored, and we have seen that in pools without any maxUnavailable value (default 1), 2 nodes start applying the OCL image at the same time.

I have a fix for this in my last push. Layered nodes were being incorrectly being marked ready for an update when they lacked the "current image" annotation on the node object(see `isNodeReady()` [call](https://github.com/openshift/machine-config-operator/pull/4489/files#diff-9842793db35f35ec7987a0daf358b0c1e90f1e065c837fe19677fa6d2027080fR381)). I cleaned up the logic to account for this case. This would only happen when the pool is first opted into layering because that is the only time the node object would be missing this annotation. So while testing, please ensure this particular scenario is being checked. Thanks!

**Comment by djoshy** [comment]
It may be worth testing https://issues.redhat.com/browse/OCPBUGS-38869 on this PR as well. I'm not seeing the issue any longer in my testing, so it is possible some of the status rework has fixed it.

File: pkg/controller/node/status.go [hunk 1]
```diff
@@ -352,12 +361,37 @@ func isNodeDone(node *corev1.Node) bool {
 		return false
 	}
 
+	if layered {
+		// The MachineConfig annotations are loaded on boot-up by the daemon which
+		// isn't currently done for the image annotations, so the comparisons here
+		// are a bit more nuanced.
+		cimage, cok := node.Annotations[daemonconsts.CurrentImageAnnotationKey]
```
Comment (yuqi-zhang) [inline pkg/controller/node/status.go:368]: Can you remind me if any of these can exist but be "empty"? For example, if the pool is opted into layering, and then opted out of layering, do these get cleaned up?

**Review by yuqi-zhang — APPROVED** [review]
Generally makes sense. 1 question inline

/hold

To give Sergio a chance to re-verify (plus the https://issues.redhat.com/browse/OCPBUGS-38869)

**Comment by sergiordlr** [comment]
Verified using IPI on AWS

We have checked that the updated nodes are correctly reported in the following scenarios:

- Apply techpreview
- Apply a new MC in no-techpreview cluster
- Apply OCL images in tech-preview

In all cases the right order was used when applying the new configs/images, and the maxUnavailable value was  honored (tested with values maxUnavailable=1 and maxUnavailable=2)

Nevertheless, https://issues.redhat.com/browse/OCPBUGS-38869 is not fixed we can still see this issue in this PR

MCN are not updated with the new config:
```
$ oc get node  -o jsonpath='{.metadata.annotations.machineconfiguration\.openshift\.io/desiredConfig}' ip-10-0-4-122.us-east-2.compute.internal
rendered-worker-048b8d2473be5ff15997817af213c185 
$ oc  get machineconfignode  -o jsonpath='{.spec.configVersion.desired}'  ip-10-0-4-122.us-east-2.compute.internal
rendered-worker-0ad05289a6c60cf6589022299b83f6f6
$ oc get mcp
NAME     CONFIG                                             UPDATED   UPDATING   DEGRADED   MACHINECOUNT   READYMACHINECOUNT   UPDATEDMACHINECOUNT   DEGRADEDMACHINECOUNT   AGE
master   rendered-master-5507a2319aefb31232b8dd16103766e4   True      False      False      3              3                   3                     0                      6h9m
worker   rendered-worker-048b8d2473be5ff15997817af213c185   True      False      False      4              4                   4                     0                      6h9m
```

/label qe-approved

File: pkg/controller/node/status.go (hunk 1 above)
Comment (sergiordlr) [reply pkg/controller/node/status.go:376]: There is no way to opt a pool out of  layering yet, right?

Comment (djoshy) [reply pkg/controller/node/status.go:376]: Not at the moment, no. There is a PR here for that piece: https://github.com/openshift/machine-config-operator/pull/4284

Comment (cheesesashimi) [reply pkg/controller/node/status.go:376]: When reverted, the `desiredImage` and `currentImage` annotations should be cleared.

**Pull request merged**